utils.py

```diff
@@ -442,6 +531,81 @@ def direction_key_pressed(button_id, n_rows, n_cols, image_list, *args):
+        elif button_id == 'move-right':
+            new_i, new_j = i, (j+1) % n_cols
+            check_class = args[N_GRID + new_j + new_i*COLS_MAX]
+        elif button_id == 'move-up':
+            new_i, new_j = (i-1) % n_rows, j
+            check_class = args[N_GRID + new_j + new_i*COLS_MAX]
+        elif button_id == 'move-down':
+            new_i, new_j = (i+1) % n_rows, j
+            check_class = args[N_GRID + new_j + new_i*COLS_MAX]
+
+        # Add focus to check_class
+        if check_class:
+            current_idx = new_i * COLS_MAX + new_j
+            new_classes[current_idx]= ' '.join(class_toggle_focus(new_classes[current_idx].split(' ')))
+            cell_last_clicked = [new_i, new_j]
+    else:
+        for i in range(ROWS_MAX):
+            for j in range(COLS_MAX):
+                my_class = args[N_GRID + j + i*COLS_MAX]
+
+                # There's no need to change the class of a cell that is hidden
+                if i >= n_rows or j >= n_cols:
+                    new_classes.append(my_class)
+                    continue
+
+                if button_id == 'move-left':
+                    right_ngbr_i, right_ngbr_j = i, (j+1) % n_cols
+                    check_class = args[N_GRID + right_ngbr_j + right_ngbr_i*COLS_MAX]
+                elif button_id == 'move-right':
+                    left_ngbr_i, left_ngbr_j = i, (j-1) % n_cols
+                    check_class = args[N_GRID + left_ngbr_j + left_ngbr_i*COLS_MAX]
+                elif button_id == 'move-up':
+                    above_ngbr_i, above_ngbr_j = (i+1) % n_rows, j
+                    check_class = args[N_GRID + above_ngbr_j + above_ngbr_i*COLS_MAX]
+                elif button_id == 'move-down':
+                    below_ngbr_i, below_ngbr_j = (i-1) % n_rows, j
+                    check_class = args[N_GRID + below_ngbr_j + below_ngbr_i*COLS_MAX]
+
+                # Move focus away from the cell with it
+                if 'focus' in my_class:
+                    new_classes.append(' '.join(class_toggle_focus(my_class.split(' '))))
+                else:
+                    # In this case, we receive focus from the appropriate neighbour:
+                    # update our class name and note the cell location for the image zoom panel
+                    # Note: as the focus was previously elsewhere, we cannot have it
+                    if 'focus' in check_class:
+                        new_classes.append(' '.join(class_toggle_focus(my_class.split(' '))))
+                        cell_last_clicked = [i, j]
+                    else:
+                        new_classes.append(my_class)
+
+    img_idx = cell_last_clicked[1] + cell_last_clicked[0]*n_cols
+    zoomed_img = html.Img(src=image_list[img_idx], style=config.IMG_STYLE_ZOOM) if len(image_list) > 0 else EMPTY_IMAGE
+    return new_classes, zoomed_img, cell_last_clicked
+
+
 def keep_delete_pressed(button_id, n_rows, n_cols, image_list, *args):
 
```

Comment:
I think this may need to be updated, as the functionality behind these buttons doesn't work any more. I will have a think.

Reply:
Fixed it!